modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js

```diff
@@ -56,6 +56,17 @@ export default function SearchBar({
 	const alignElementRef = useRef();
 	const dropdownRef = useRef();
 
+	const characterThresholds = JSON.parse(suggestionsContributorConfiguration)
+		.filter((config) => config.attributes?.characterThreshold)
+		.map((config) => parseInt(config.attributes.characterThreshold, 10));
+
+	// If the blueprint suggestions contributor does not set its own threshold,
+	// the suggestions display threshold will default to the global one.
+
+	const finalSuggestionsDisplayThreshold = characterThresholds.length
+		? Math.min(...characterThresholds)
+		: parseInt(suggestionsDisplayThreshold, 10);
```

Comment:
2 suggestions:
1) Instead of having these variables that will get re-computed every re-render, we should move this to a function with `useCallback`.
2) Instead of the name `finalSuggestionsDisplayThreshold` which seems very vague to me (specifically the word "final" is vague), we can use a name more related to the purpose. Like `_getLowestSuggestionsDisplayThreshold`.